sklearn/linear_model/ridge.py

```diff
@@ -545,6 +545,13 @@ def fit(self, X, y, sample_weight=None):
                          accept_sparse=_accept_sparse,
                          dtype=_dtype,
                          multi_output=True, y_numeric=True)
+        if sparse.issparse(X) and self.fit_intercept:
+            solver = 'sparse_cg'
+            if self.solver not in ['auto', 'sparse_cg']:
+                warnings.warn(
+                    'setting solver to "sparse_cg" because X is sparse')
```

Comment:
Better,
```
"solver={} does not support fitting the intercept on sparse data, "
"falling back solver='sparse_cg'. To avoid this warning either change the solver "
"to 'sparse_cg' explicitly or set `fit_intercept=False`.
```